AutotypeURL/ATUAutotypeURL.m

```diff
@@ -51,14 +52,33 @@ - (nonnull NSString *)windowTitleForRunningApplication:(nonnull NSRunningApplica
 
 - (NSString *)_getURLFromSafari {
   NSAppleScript *script = [[NSAppleScript alloc] initWithSource:@"tell application \"Safari\" to get URL of front document"];
-  NSAppleEventDescriptor *aed = [script executeAndReturnError:NULL];
-  return aed.stringValue;
+    NSAppleEventDescriptor *aed = [script executeAndReturnError:NULL];
+    NSString *rawURL = aed.stringValue;
+  
+
+    NSLog(@"%@ - raw URL", rawURL);
+
+    NSLog(@"%@ - parsedURL", [self _parseURL:rawURL]);
+    return [self _parseURL:rawURL];
+    
+    
 }
 
 - (NSString *)_getURLFromChrome {
   NSAppleScript *script = [[NSAppleScript alloc] initWithSource:@"tell application \"Google Chrome\" to get URL of active tab of front window"];
   NSAppleEventDescriptor *aed = [script executeAndReturnError:NULL];
   return aed.stringValue;
+}
+//Additions for parsing the url subdomain.root.com
+- (NSString *)_parseURL: (NSString*) url;
+{
+    NSArray *separatedURL = [url componentsSeparatedByCharactersInSet:[NSCharacterSet characterSetWithCharactersInString:@"/"]];
```

Comment:
Just a guess: Could you use ```NSURL``` for a more robust parsing? I would guess that the URL field always returns a valid URL so ```[[NSURL alloc] initWithString:url]``` should be useable.

Reply:
done. i was able to get this working with out too much issue.
thanks for the recommdation. I used an array cause thats how my AppleScript worked, but you are right this is far more robust.